### DISCUSSION TITLE
[A/L] (2021/22) Funzioni - Esercizio 17

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#257] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152315888-906e24ed-2f8b-4daf-affe-b947b5fa3cf2.png)

**Comment by LuginiAndrea** [comment]
La risposta è si.
Infatti, prendendo 

- $f: -\mathbf{N} \rightarrow \mathbf{N}, f\left(x\right) = x^2$
- $g : \mathbf{N} \rightarrow \mathbf{N}, g\left(z\right) = z^2$
Dimostriamo per assurdo che $f$ (ed implicitamente anche $g$) è iniettiva (dimostriamo quindi che non può essere non iniettiva):
$$x_1 \neq x_2, f\left(x_1\right) = f\left(x_2\right) \Longrightarrow x_1^2 = x_2^2 \Longrightarrow \pm \sqrt[2]{x_1^2} = \pm\sqrt[2]{x_2^2}, \textrm{ma ignoriamo le radici positive quanto il dominio di $f$ è quello dei naturali negativi } \Longrightarrow -\sqrt[2]{x_1^2} = -\sqrt[2]{x_2^2} \Longrightarrow x_1 = x_2$$
$f$ è quindi iniettiva.

Consideriamo ora $h$, che è definita così:
$$h: \left(-\mathbf{N} \cup \mathbf{N}\right) \rightarrow \mathbf{N}, h\left(x\right) = x^2$$
Essendo $-\mathbf{N} \cup \mathbf{N} = \mathbf{Z}$
Allora
$$h: \left(-\mathbf{N} \cup \mathbf{N}\right) \rightarrow \mathbf{N} \Longrightarrow h:\mathbf{Z} \rightarrow \mathbf{N}$$
Possiamo vedere subito che $h$ non è iniettiva. Infatti prendendo $x_1 = -2$ e $x_2 = 2$ possiamo notare che
$$-2 \neq 2, \left(-2\right)^2 = 2^2$$.

Ricapitolando, abbiamo dimostrato che possono esistere $f$ ed $g$ iniettive con $h$ non iniettiva
